frontend/src/redux/Game.ts

```diff
@@ -0,0 +1,29 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Game, getGame } from '../api/Game';
+
+const initialState = null as Game | null;
+
+export const fetchGame = createAsyncThunk<Game, string>(
+  'games/fetch',
+  async (roomId, thunkApi) => getGame(roomId)
+    .then((res) => res)
+    .catch((err) => thunkApi.rejectWithValue(err)),
```

Comment:
I think pretty much, with the exception that you'll need to add a `.then(unwrapResult)` beforehand (which is a method coming from the redux toolkit library)

Actually I think it should be `thunkApi.rejectWithValue(axiosErrorHandler(err))` so I'll update that

Reply:
Actually ignore that last comment - I think it's correct as is haha - but yeah it does pretty much just pass along the error so that you can deal with it in a catch block in other files similar to what we currently do